packages/astro/src/assets/build/remote.ts

```diff
@@ -0,0 +1,55 @@
+import CachePolicy from 'http-cache-semantics';
+
+export type RemoteCacheEntry = { data: string; expires: number };
+
+export async function loadRemoteImage(src: string) {
+	try {
+		if (src.startsWith('//')) {
+			src = `https:${src}`;
```

Comment:
I don't think we should prefix the protocol. Many developers use `//` on purpose because internal environments (`local`, `dev`, `test`, `staging`, `pre-prod`, etc.) might not be under `https`.

Reply:
Well, the problem is that there's no way for us to know the protocol from here, we have to assume one

Reply:
I'm not very familiar with the business logic, but why do we need to prepend the https?

Reply:
To actually fetch the remote image 😄 In SSR, it's totally fine for users to do `//...` because we can assume the protocol from the current request, however in SSG, `//` means nothing, because there's no request to infer it from.

Doing `fetch("//...")` will throw with an invalid URL (of course), so we need to prepend a protocol.

Reply:
Oh, I see. It's a gamble. Shouldn't we try to fetch the image with both protocols?